**CASE STUDY: OPTIMIZING LANDING PAGE VARIANTS FOR REVENUE**

**WHAT WE TESTED:**
three landing page designs (Vibrant, Cold, Heat) across 30,000 user sessions
for our Bluetooth speakers.

**KEY OBJECTIVES:**

*   Compare conversion and purchase rates across Vibrant, Cold, and Heat.
*   See how user engagement (time spent, pages visited) relates to checkouts.
*   Find out if specific segments (traffic sources, devices) prefer certain designs.
*   Calculate revenue per session to ensure we are maximizing profit, not just signups.


**STEPS:**

**Step 1-Statistical Testing:**

Run a Chi-Square test to check if conversion rate differences are statistically significant (p < 0.05) and use ANOVA to check differences in revenue and time spent.

**Step 2-Funnel & Behavior:**

Filter out bounces to analyze pages visited and time spent. See which variant keeps users engaged longer.

**Step 3-Segmentation:**

Cross-tabulate variants with traffic sources and device types to spot hidden preference trends.

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats
import statsmodels.api as sm

In [ ]:
df = pd.read_csv("speakers_data.csv")

In [ ]:
df.head(10)

,user_id,session_id,sign_in,name,demographic_age,demographic_age_group,demographic_gender,email,location,country,...,pages_visited,conversion_flag,conversion_type,traffic_source,product_purchased,revenue_$,payment_type,card_type,coupon_applied,bounce_flag
0,U10477,S000001,Email,Victor Navarro-Noël,31,Adult,Female,victornavarronoël251@hotmail.com,Rome,Italy,...,7,0,NCT,Organic,NPP,0.00,NPT,NCAT,ND,1
1,U01536,S000002,Email,王秀云,39,Adult,Female,王秀云617@gmail.com,Madrid,Spain,...,5,0,NCT,Social,NPP,0.00,NPT,NCAT,ND,0
2,U00107,S000003,Guest,Ucchal Sabharwal,68,Old,Male,Not Provided,Manchester,UK,...,7,0,NCT,Organic,NPP,0.00,NPT,NCAT,ND,0
3,U13886,S000004,Email,Virginie Schmitt,72,Old,Female,virginieschmitt827@gmail.com,Sydney,Australia,...,10,0,NCT,Social,NPP,0.00,NPT,NCAT,ND,0
4,U05926,S000005,Email,Cynthia Drake,51,Adult,No Answer,cynthiadrake47@hotmail.com,Mumbai,India,...,6,0,NCT,Organic,NPP,0.00,NPT,NCAT,ND,0
5,U05821,S000006,Email,Laure-Suzanne Durand,35,Adult,Female,lauresuzannedurand719@gmail.com,Mumbai,India,...,9,0,NCT,Organic,NPP,0.00,NPT,NCAT,ND,1
6,U05314,S000007,Guest,Rachel Medina DVM,43,Adult,Female,Not Provided,Shanghai,China,...,10,1,Purchase,Social,Marshall Kilburn II,299.99,Card,Visa,No,0
7,U14984,S000008,Email,Damyanti Balasubramanian,45,Adult,Female,damyantibalasubramanian272@hotmail.com,Munich,Germany,...,10,0,NCT,Organic,NPP,0.00,NPT,NCAT,ND,0
8,U03594,S000009,Guest,Lipika Kata,79,Old,Female,Not Provided,Beijing,China,...,3,0,NCT,Social,NPP,0.00,NPT,NCAT,ND,0
9,U06917,S000010,Email,山本 亮介,73,Old,Female,山本亮介391@protonmail.com,Munich,Germany,...,2,0,NCT,Paid,NPP,0.00,NPT,NCAT,ND,0


**statistical test to evaluate the A/B/C test performance of those landing page variants (Vibrant, Cold, Heat).**

In [ ]:
print("Dataset Shape:", df.shape)

# 1. Basic Inspection: Check missing values and data types
print("\n--- Missing Values Check ---")
print(df.isnull().sum()[df.isnull().sum() > 0])

# 2. A/B/C Test Performance: Conversion Rate by Variant Group
print("\n--- Conversion Rate & Revenue by Landing Page Variant ---")
variant_summary = df.groupby('variant_group').agg(
    total_sessions=('session_id', 'count'),
    total_conversions=('conversion_flag', 'sum'),
    conversion_rate=('conversion_flag', 'mean'),
    avg_revenue=('revenue_$', 'mean'),
    total_revenue=('revenue_$', 'sum')
).reset_index()

# Convert conversion rate to percentage for readability
variant_summary['conversion_rate (%)'] = (variant_summary['conversion_rate'] * 100).round(2)
display(variant_summary)

# 3. Statistical Significance Test (Chi-Square Test for Independence)
print("\n--- Chi-Square Test: Variant Group vs. Conversion Flag ---")
contingency_table = pd.crosstab(df['variant_group'], df['conversion_flag'])
chi2, p_val, dof, expected = stats.chi2_contingency(contingency_table)

print(f"Chi2 Statistic: {chi2:.4f}")
print(f"p-value: {p_val:.4f}")

if p_val < 0.05:
    print("Result: Statistically significant difference in conversion rates across variants! (Reject Null Hypothesis)")
else:
    print("Result: No statistically significant difference in conversion rates across variants. (Fail to Reject Null Hypothesis)")

Dataset Shape: (30000, 24)

--- Missing Values Check ---
Series([], dtype: int64)

--- Conversion Rate & Revenue by Landing Page Variant ---


,variant_group,total_sessions,total_conversions,conversion_rate,avg_revenue,total_revenue,conversion_rate (%)
0,Cold,9928,1795,0.180802,30.740110,305187.81,18.08
1,Heat,10026,1521,0.151706,26.181090,262491.61,15.17
2,Vibrant,10046,1219,0.121342,21.757632,218577.17,12.13



--- Chi-Square Test: Variant Group vs. Conversion Flag ---
Chi2 Statistic: 137.6157
p-value: 0.0000
Result: Statistically significant difference in conversion rates across variants! (Reject Null Hypothesis)


In [ ]:
print(df.columns)

Index(['user_id', 'session_id', 'sign_in', 'name', 'demographic_age',
       'demographic_age_group', 'demographic_gender', 'email', 'location',
       'country', 'device_type', 'timestamp', 'variant_group', 'time_spent',
       'pages_visited', 'conversion_flag', 'conversion_type', 'traffic_source',
       'product_purchased', 'revenue_$', 'payment_type', 'card_type',
       'coupon_applied', 'bounce_flag'],
      dtype='object')



Cold Variant: 18.08% conversion rate,

Avg Revenue: $ 30.74, Total Revenue: $ 305K

Heat Variant: 15.17% conversion rate,

Avg Revenue: $ 26.18, Total Revenue: $ 262K

Vibrant Variant: 12.13% conversion rate,

Avg Revenue: $ 21.76, Total Revenue: $ 218K


The **Cold variant** is completely dominating the other variants across both conversion rate and average transaction value.


In [ ]:
# 1. Behavioral Engagement by Variant Group
print("--- Behavioral Engagement by Variant ---")
engagement_summary = df.groupby('variant_group').agg(
    avg_time_spent=('time_spent', 'mean'),
    avg_pages_visited=('pages_visited', 'mean'),
    bounce_rate=('bounce_flag', 'mean')
).reset_index()

engagement_summary['bounce_rate (%)'] = (engagement_summary['bounce_rate'] * 100).round(2)
display(engagement_summary)

# 2. Conversion Rate by Traffic Source & Variant Group
print("\n--- Conversion Rate by Traffic Source & Variant ---")
traffic_variant = df.groupby(['variant_group', 'traffic_source']).agg(
    sessions=('session_id', 'count'),
    conversion_rate=('conversion_flag', 'mean'),
    avg_revenue=('revenue_$', 'mean')
).reset_index()

traffic_variant['conversion_rate (%)'] = (traffic_variant['conversion_rate'] * 100).round(2)
display(traffic_variant)

--- Behavioral Engagement by Variant ---


,variant_group,avg_time_spent,avg_pages_visited,bounce_rate,bounce_rate (%)
0,Cold,10.534523,5.492748,0.186644,18.66
1,Heat,10.481272,5.501596,0.190604,19.06
2,Vibrant,10.556797,5.509954,0.189926,18.99



--- Conversion Rate by Traffic Source & Variant ---


,variant_group,traffic_source,sessions,conversion_rate,avg_revenue,conversion_rate (%)
0,Cold,Organic,4892,0.177433,30.589074,17.74
1,Cold,Paid,2012,0.184394,27.786889,18.44
2,Cold,Referral,991,0.198789,40.455267,19.88
3,Cold,Social,2033,0.176586,29.290541,17.66
4,Heat,Organic,5074,0.153331,27.459493,15.33
5,Heat,Paid,2049,0.162518,27.098370,16.25
6,Heat,Referral,958,0.134656,21.572777,13.47
7,Heat,Social,1945,0.144473,24.149542,14.45
8,Vibrant,Organic,5118,0.124853,24.218515,12.49
9,Vibrant,Paid,1889,0.121228,20.490561,12.12


In [ ]:
#Funnel Analysis

# Funnel Analysis by Variant Group
print("--- Funnel Analysis by Landing Page Variant ---")

# We create step flags for clarity
df['engaged_flag'] = 1 - df['bounce_flag'] # 1 if they didn't bounce
df['signed_in_flag'] = np.where(df['sign_in'] == 'Email', 1, 0)

funnel_summary = df.groupby('variant_group').agg(
    total_sessions=('session_id', 'count'),
    engaged_sessions=('engaged_flag', 'sum'),
    signed_in_users=('signed_in_flag', 'sum'),
    converted_users=('conversion_flag', 'sum')
).reset_index()

# Calculate drop-off/conversion rates from total sessions
funnel_summary['engagement_rate (%)'] = (funnel_summary['engaged_sessions'] / funnel_summary['total_sessions'] * 100).round(2)
funnel_summary['signin_rate (%)'] = (funnel_summary['signed_in_users'] / funnel_summary['total_sessions'] * 100).round(2)
funnel_summary['overall_conversion_rate (%)'] = (funnel_summary['converted_users'] / funnel_summary['total_sessions'] * 100).round(2)

display(funnel_summary[['variant_group', 'total_sessions', 'engaged_sessions', 'signed_in_users', 'converted_users', 'overall_conversion_rate (%)']])

--- Funnel Analysis by Landing Page Variant ---


,variant_group,total_sessions,engaged_sessions,signed_in_users,converted_users,overall_conversion_rate (%)
0,Cold,9928,8075,6921,1795,18.08
1,Heat,10026,8115,7045,1521,15.17
2,Vibrant,10046,8138,7009,1219,12.13


# Insights & Recommendations

**User engagement is actually identical across all three pages.** People spend about 10.5 minutes, browse 5.5 pages, and bounce 18 to 19 percent of the time no matter which design they look at.

That means the designs aren't changing how long people stay or browse. Instead, the **Cold variant simply removes friction at the final decision point** and drives higher purchase intent.

Even better, **Cold wins across every single traffic source**—whether it's organic, paid, social, or referral. Referral traffic is a massive standout here, hitting a **19.88 percent conversion rate** and a top basket size of **$40.45 under the Cold variant**.

As a growth analyst, here is our bottom line for leadership:

* **Action:** Stop the Vibrant and Heat tests right now and roll out the Cold variant to 100 percent of traffic.

* **Financial Forecast:** Scaling the Cold design which pulled in around $ 305K during the test will maximize total revenue and lift our overall conversion rate from **12 to 15 percent up to a solid 18 percent** **(moving from 12.13% to 18.08% is roughly 50% boost-over $135,000 in additional revenue)**

* **Strategic Growth Pivot:** Double down on referral partnerships. When you pair them with the optimized Cold design, they yield our **highest revenue per user at $40.45.**


### Key Financial & Conversion Metrics

* **Conversion Rate Lift:** The **Cold** variant achieved an 18.08% conversion rate, significantly outperforming *Heat* (15.17%) and *Vibrant* (12.13%). Moving from the lowest-performing variant to the highest represents a **49% relative increase in conversion efficiency**.
* **Revenue Per Session (ARPU):** Users landing on the *Cold* page generated an average of **$30.74 per session**, compared to $26.18 for *Heat* and $21.76 for *Vibrant*.
* **Current Blended Performance:**
Across the test period's split traffic of 30,000 sessions, the site generated 4,535 total conversions and **$786,256** in total revenue.

---

### Projected Financial Impact of Scaling the Winner

If leadership chooses to conclude the test and route 100% of incoming website traffic to the **Cold** landing page variant, the projected performance scales as follows:

* **Projected Total Conversions:** **5,424 paying customers** (an immediate net gain of 889 additional conversions from the exact same traffic volume).
* **Projected Total Revenue:** **$922,200** (up from the current $786,256).
* **Net Revenue Lift:** An incremental **+$135,944** in top-line revenue generated purely through design optimization rather than increased ad spend.

---

### Strategic Recommendations for Stakeholders

1. **Immediate Rollout:** Terminate the *Vibrant* and *Heat* landing page tests immediately and transition 100% of web traffic to the **Cold** design layout to capture the projected revenue lift.
2. **Channel Synergy:** The data indicates that **Referral traffic** yields the highest average order value ($40.45) when paired with the *Cold* variant. Marketing operations should prioritize partnerships and affiliate channels that drive referral traffic.
3. **Investigate Friction Points:** While top-of-funnel engagement and sign-in rates remained identical (~80% engagement) across all three pages, the *Cold* variant successfully eliminated checkout friction. Product and UX teams should audit the final checkout modal of the *Cold* page to identify what psychological cues drove that final purchase intent, and apply those learnings to future product updates.